monad-logger/Control/Monad/Logger.hs

```diff
@@ -364,12 +364,15 @@ logOtherS = [|\src level msg -> monadLoggerLog $(qLocation >>= liftLoc) src (Lev
 -- Since 0.2.4
 newtype NoLoggingT m a = NoLoggingT { runNoLoggingT :: m a }
 
-instance Monad m => Functor (NoLoggingT m) where
```

Comment:
This change in constraints is a non-breaking change since the AMP proposal went through (GHC 7.10?), but represents a breaking API change for earlier GHCs. I think it's better to only introduce the `Functor`/`Applicative` constraints conditionally for GHC 7.10 and newer.